querybook/server/lib/vector_store/base_vector_store.py

```diff
@@ -39,25 +52,54 @@ def should_skip_query_execution(
 
         return False
 
-    def search_tables(self, text: str, threshold=0.6, k=3):
+    def search_tables(
+        self,
+        metastore_id: int,
+        text: str,
+        search_type: Optional[Literal["table", "query"]] = None,
+        threshold=0.6,
+        k=DEFAUTL_TABLE_SEARCH_LIMIT,
+        fetch_k=DEFAULT_VECTOR_STORE_FETCH_LIMIT,
+    ) -> list[tuple[str, int]]:
         """Find tables using embedding based table search.
 
         The table search will return a list of k tables that are similar to the given text with highest similarity score.
+
+        Args:
+            metastore_id: the metastore id
+            text: the text to search
+            search_type: the type of the text. It can be "table" or "query" or None. If it is None, it will search both tables and queries summary.
+            threshold: the threshold of the similarity score. Only return tables with score higher than the threshold.
+            k: the number of tables to return.
+            fetch_k: the number of tables to fetch from the vector store.
+
+        Returns:
+            a list of tuples (table_name, score)
         """
-        docs_with_score = self.similarity_search_with_score(text, k=3)
+
+        must_query = [{"term": {"metadata.metastore_id": metastore_id}}]
+        if search_type:
+            must_query.append({"term": {"metadata.type": search_type}})
+        boolean_filter = {"bool": {"must": must_query}}
+
+        docs_with_score = self.similarity_search_with_score(
+            text, k=fetch_k, boolean_filter=boolean_filter
+        )
         tables = [
-            (table, score)
+            (table_name, score, doc.metadata.get("type"))
             for (doc, score) in docs_with_score
-            for table in doc.metadata.get("tables", [])
+            for table_name in doc.metadata.get("tables", [])
             if score > threshold
         ]
 
         table_score_dict = {}
-        for table, score in tables:
-            table_score_dict[table] = max(score, table_score_dict.get(table, 0))
+        for table_name, score, type in tables:
+            # TODO: need to tune the scoring strategy
+            if type == "table" and score >= 0.7:
```

Comment:
how about we use (threshold + 0.1) to prevent there are too many magic constants